src/main/java/Duke.java

```diff
@@ -1,81 +1,200 @@
+import java.util.ArrayList;
 import java.util.Scanner;
 
 public class Duke {
 
-    private static Task[] taskList = new Task[100];
-    private static int taskCount = 0;
+    private static final int SECTION_LIMIT = 2;
 
-    private static final int MARK_NUMBER_INDEX = 5;
-    private static final int NEWLINE_STRIP = 2;
-    private static final int UNMARK_NUMBER_INDEX = 7;
+    private static ArrayList<String> responses = new ArrayList<>();
+    private static int taskCount = 0;
+    private static Task[] taskList = new Task[100];
 
     public static void main(String[] args) {
-        botResponse("Hello! I'm Boba\n\tWhat can I do for you?");
+        responses.add("Hello! I'm Boba.");
+        responses.add("I am a bot 'tasked' to manage your tasks");
+        responses.add("What can I do for you?");
+        responses.add("Type 'help' to get the list commands I response to");
+        botResponse();
         Scanner scan = new Scanner(System.in);
         String input = scan.nextLine();
-        while (!input.equals("bye")) {
-            if (input.equals("list")) {
-                // StringBuilder to reduce constructing a new string due to concatenation
-                StringBuilder sb = new StringBuilder();
-                for (int i = 0; i < taskCount; i++) {
-                    sb.append(i + 1); // Post-increment to keep index up to date
-                    sb.append(". ");
-                    sb.append(taskList[i].toString());
-                    sb.append("\n\t");
+        Command operation = getCommand(input);
+        String[] sections = parseInput(operation, input);
+        while (operation != Command.EXIT) {
+            if (checkNull(sections)) {
+                printInvalidCommand(operation);
+            } else {
+                switch (operation) {
+                case LIST:
+                    printTasks();
+                    break;
+                case MARK:
+                    markTask(true, sections[0], operation);
+                    break;
+                case UNMARK:
+                    markTask(false, sections[0], operation);
+                    break;
+                case TODO:
+                    addTask(new Todo(sections[0]));
+                    break;
+                case DEADLINE:
+                    addTask(new Deadline(sections[0], sections[1]));
+                    break;
+                case EVENT:
+                    addTask(new Event(sections[0], sections[1]));
+                    break;
+                case HELP:
+                    printHelp();
+                    break;
+                case NONE:
+                    printInvalidCommand(operation);
```

Comment:
I think can add this switch function into a new class named taskManager.

Reply:
Thank you for the suggestion! I agree a new class like taskManager would be a good addition